public/resources/core/js/controllers/product-offering.controller.js

```diff
@@ -316,29 +285,73 @@
         }
     }
 
-    function OfferingUpdateController($state, $rootScope, EVENTS, Offering, Utils) {
+    function ProductOfferingUpdateController($state, $scope, $rootScope, $controller, EVENTS, DATA_STATUS, Offering, Utils) {
         /* jshint validthis: true */
         var vm = this;
-        var initialData = {};
-        var pachable = ['name', 'version', 'description', 'lifecycleStatus'];
+
+        angular.extend(vm, $controller('FormMixinCtrl', {$scope: $scope}));
+
+        $scope.CHARGE_PERIODS = Offering.TYPES.CHARGE_PERIOD;
+        $scope.CURRENCY_CODES = Offering.TYPES.CURRENCY_CODE;
+        $scope.PRICES = Offering.TYPES.PRICE;
 
         vm.update = update;
         vm.updateStatus = updateStatus;
         vm.hasCategory = hasCategory;
 
-        vm.item = {};
+        vm.priceplan = new Offering.Priceplan();
 
-        Offering.detail($state.params.offeringId).then(function (offeringRetrieved) {
-            initialData = angular.copy(offeringRetrieved);
-            vm.data = angular.copy(offeringRetrieved);
-            vm.item = offeringRetrieved;
-            vm.item.status = LOADED;
-            vm.categories = vm.item.getCategories();
-        }, function (reason) {
-            vm.error = Utils.parseError(reason, 'The requested offering could not be retrieved');
-            vm.item.status = ERROR;
+        vm.createPriceplan = createPriceplan;
+        vm.updatePriceplan = updatePriceplan;
+        vm.removePriceplan = removePriceplan;
+
+        vm.status = DATA_STATUS.LOADING;
+
+        $scope.$on(Offering.EVENTS.PRICEPLAN_UPDATED, function (event, priceplan) {
```

Comment:
Do you really need this event and having a separate controller for the price plan that basically triggers this event?